src/System.IO.Pipelines/ReadCursor.cs

```diff
@@ -209,6 +242,52 @@ internal bool TryGetBuffer(ReadCursor end, out Memory<byte> data, out ReadCursor
             }
         }
 
+        private bool TryGetBufferMultiBlock(ReadCursor end, out Memory<byte> data)
+        {
+            var segment = _segment;
+            var index = _index;
+
+            // Determine if we might attempt to copy data from segment.Next before
```

Comment:
This comment isn't quite true anymore. Pipelines takes a snapshot of start/end before these methods navigate the linked list.